flink-table-store-core/src/main/java/org/apache/flink/table/store/file/operation/FileStoreWriteImpl.java

```diff
@@ -0,0 +1,107 @@
+import org.apache.flink.table.store.file.utils.RecordWriter;
+import org.apache.flink.table.types.logical.RowType;
+
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.concurrent.ExecutorService;
+import java.util.stream.Collectors;
+
+/** Default implementation of {@link FileStoreWrite}. */
+public class FileStoreWriteImpl implements FileStoreWrite {
+
+    private final RowType keyType;
+    private final RowType rowType;
+    private final Comparator<RowData> keyComparator;
+    private final Accumulator accumulator;
+    private final FileFormat fileFormat;
+    private final FileStorePathFactory pathFactory;
+    private final MergeTreeOptions mergeTreeOptions;
+
+    private final FileStoreScan scan;
+
+    public FileStoreWriteImpl(
+            RowType partitionType,
+            RowType keyType,
+            RowType rowType,
+            Comparator<RowData> keyComparator,
+            Accumulator accumulator,
+            FileFormat fileFormat,
+            FileStorePathFactory pathFactory,
+            MergeTreeOptions mergeTreeOptions) {
+        this.keyType = keyType;
+        this.rowType = rowType;
+        this.keyComparator = keyComparator;
+        this.accumulator = accumulator;
+        this.fileFormat = fileFormat;
+        this.pathFactory = pathFactory;
+        this.mergeTreeOptions = mergeTreeOptions;
+
+        this.scan = new FileStoreScanImpl(partitionType, keyType, rowType, fileFormat, pathFactory);
+    }
+
+    @Override
+    public RecordWriter createWriter(
+            BinaryRowData partition, int bucket, ExecutorService compactExecutor) {
+        Long latestSnapshotId = pathFactory.latestSnapshotId();
+        if (latestSnapshotId == null) {
+            return createEmptyWriter(partition, bucket, compactExecutor);
+        } else {
+            MergeTree mergeTree = createMergeTree(partition, bucket, compactExecutor);
+            return mergeTree.createWriter(
+                    scan.withSnapshot(latestSnapshotId)
+                            .withPartitionFilter(Collections.singletonList(partition))
+                            .withBucket(bucket).plan().files().stream()
+                            .map(ManifestEntry::file)
+                            .collect(Collectors.toList()));
+        }
+    }
+
+    @Override
+    public RecordWriter createEmptyWriter(
+            BinaryRowData partition, int bucket, ExecutorService compactExecutor) {
+        MergeTree mergeTree = createMergeTree(partition, bucket, compactExecutor);
+        return mergeTree.createWriter(Collections.emptyList());
+    }
+
+    private MergeTree createMergeTree(
+            BinaryRowData partition, int bucket, ExecutorService compactExecutor) {
+        SstFile sstFile =
+                new SstFile(
+                        keyType,
+                        rowType,
+                        fileFormat,
+                        pathFactory.createSstPathFactory(partition, bucket),
```

Comment:
> Can uuid be reused in SstPathFactory from FileStorePathFactory?
 
Any update?

Reply:
Why are we reusing this? One of the benefits I can see is that files written by the same job will contain the same uuid but currently I don't think this is a must.